### PR TITLE
feat: dynamic max threshold customization

### PR DESCRIPTION
**Pull request by ryzzaki**
## Description
This PR addresses issue #214

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by prokopsimek** [comment]
# [DX Scanner](https://dxscanner.io) Report <a href="https://dxscanner.io" target="_blank">DXScanner.io</a>
## https://github.com/DXHeroes/dx-scanner 

<h3>🚨 Improvements with highest impact
</h3>
<details>
<summary>Break down large pull requests into smaller ones</summary>

<b>Large pull request are hard to code review and it reduces the probability of finding bugs. Split your PRs into logical units. Do not have PR with more than 1000 changes.</b> <a href="https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067">https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067</a>
  <b>Impact changed from medium to high.</b>
</details>
<h3>⚠️ Improvements with medium impact
</h3>
<details>
<summary>Solve Issues Continuously</summary>

<b>Do not have an open Issues more than 60 days. Solve Issues continuously.</b> <a href="https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612">https://hackernoon.com/45-github-issues-dos-and-donts-dfec9ab4b612</a>
</details>
<details>
<summary>Update Dependencies of Minor and Patch Level</summary>

<b>Keep the dependencies updated to eliminate security concerns and compatibility issues. Use, for example, npm-check-updates.</b> <a href="https://dxkb.io/p/updating-the-dependencies">https://dxkb.io/p/updating-the-dependencies</a>


 Name              | New    | Current  
------------------- | -------- | ----------
 npm-check-updates | 4.0.3  | 4.0.1    
 tslib             | 1.11.1 | 1.11.0   
 typescript        | 3.8.3  | 3.8.2    
 @types/node       | 13.7.7 | 12.12.28 
 nock              | 12.0.2 | 12.0.1   
 @types/node       | 13.7.7 | 12.12.28 

  <b>Impact changed from high to medium.</b>
</details>
<h3>🔔 Improvements with minor impact
</h3>
<details>
<summary>Write Commit Messages by Convention</summary>

<b></b> <a href="https://www.conventionalcommits.org/">https://www.conventionalcommits.org/</a>


 Commit Message                | Problems                                        
------------------------------- | -------------------------------------------------
 Update scanner test timeouts  | subject may not be empty; type may not be empty 
 Update readme with fixer info | subject may not be empty; type may not be empty 
 Add dxs config fix flag tests | subject may not be empty; type may not be empty 
 Add fix flag to dxs config    | subject may not be empty; type may not be empty 

</details>
<details>
<summary>Update Dependencies of Major Level</summary>

<b>Keep the dependencies updated to have all possible features. Use, for example, npm-check-updates.</b> <a href="https://dxkb.io/p/updating-the-dependencies">https://dxkb.io/p/updating-the-dependencies</a>


 Name        | New    | Current  
------------- | -------- | ----------
 @types/node | 13.7.7 | 12.12.28 

</details>
<details>
<summary>Use a JS Logging Library</summary>

<b>Use a logging library to avoid errors and even cyber attacks. The most widely used logging library in the JavaScript community is Winston.</b> <a href="https://www.npmjs.com/package/winston/">https://www.npmjs.com/package/winston/</a>
</details>


---



Implementation is not adoption.
We can help you with both. :-)
[dxheroes.io](https://dxheroes.io)



##### Found a bug? Please <a href="https://github.com/DXHeroes/dx-scanner/issues" target="_blank">report</a>.

File: README.md [hunk 1]
```diff
@@ -209,6 +209,20 @@ Example `dxscannerrc.json`:
 }
 ```
 
+In order to override the default values of certain practices, specify `override` and nest the correct overridable variables.
```
Comment (prokopsimek) [inline README.md]: It would be great to have JSON:YAML example 1:1.

File: .dxscannerrc.yaml [hunk 2]
```diff
@@ -1,2 +1,6 @@
 practices:
   JavaScript.DependenciesVersionMinorPatchLevel: medium
+  LanguageIndependent.ThinPullRequestsPractice:
```
Comment (prokopsimek) [inline .dxscannerrc.yaml]: Remove?

File: .dxscannerrc.yaml [hunk 3]
```diff
@@ -1,2 +1,2 @@
 practices:
-  JavaScript.DependenciesVersionMinorPatchLevel: medium
+  JavaScript.DependenciesVersionMinorPatchLevel: medium
```
Comment (adelkahomolova) [inline .dxscannerrc.yaml]:
```suggestion
  JavaScript.DependenciesVersionMinorPatchLevel: medium
  
```

Comment (ryzzaki) [reply .dxscannerrc.yaml]: @adelkahomolova thank you for the review, unfortunately I don't understand German. Maybe @prokopsimek can assist me?

Comment (prokopsimek) [reply .dxscannerrc.yaml]: @ryzzaki Me neither. 🤪
